spec/helper/MockHelper.ts

```diff
@@ -0,0 +1,318 @@
+import {
+  AdBreakEvent,
+  AdEvent,
+  AirplayChangedEvent,
+  ErrorEvent,
+  LinearAd,
+  PlaybackEvent,
+  PlayerAPI,
+  PlayerEvent,
+  PlayerEventBase,
+  PlayerEventCallback,
+  SeekEvent,
+  TimeShiftEvent,
+  VideoPlaybackQualityChangedEvent,
+} from 'bitmovin-player';
+
+declare const global: any;
+export namespace MockHelper {
+  export function getPlayerMock(): TestingPlayerAPI {
+    const eventHelper = new PlayerEventHelper();
+
+    const PlayerMockClass: jest.Mock<TestingPlayerAPI> = jest.fn().mockImplementation(() => {
+      return {
+        ads: jest.fn(),
+        getSource: jest.fn(),
+        exports: {
+          PlayerEvent,
+          ViewMode: {
+            Fullscreen: 'fullscreen',
+          },
+        },
+        isLive: jest.fn(),
+        getConfig: jest.fn(() => {
+          return {};
+        }),
+        isPlaying: jest.fn(),
+        isPaused: jest.fn(),
+        isAirplayActive: jest.fn(),
+        isAirplayAvailable: jest.fn(),
+        getPlayerType: jest.fn(),
+        getStreamType: jest.fn(),
+        getDuration: jest.fn(),
+        getVolume: jest.fn(),
+        getContainer: jest.fn(() => {
+          return document.getElementById('player');
+        }),
+        getViewMode: jest.fn(),
+        hasEnded: jest.fn(),
+        isStalled: jest.fn(),
+        isCasting: jest.fn(),
+
+        // Event faker
+        eventEmitter: eventHelper,
+        on: eventHelper.on.bind(eventHelper),
+      };
+    });
+
+    return new PlayerMockClass();
+  }
+}
+
+export interface TestingPlayerAPI extends PlayerAPI {
+  eventEmitter: EventEmitter;
+}
+
+interface EventEmitter {
```

Comment:
What's the purpose of this interface if there's only one class that implements it?

Reply:
Good question ... actually nothing :D